coremltools/models/neural_network/builder.py

```diff
@@ -3298,6 +3297,22 @@ def set_pre_processing_parameters(self, image_input_names=None, is_bgr=False,
         if not isinstance(image_scale, dict):
             image_scale = dict.fromkeys(image_input_names, image_scale)
 
+        # Raise error if any key in image preprocessing parameters
+        # are not in imaghe_input_names.
+        def check_valid_preprocessing_keys(input, target, input_name):
+            for key in input:
+                if not key in target:
+                    raise ValueError('Invalid key {} in {}.'.format(key, input_name))
+
+        target = image_input_names
+        check_valid_preprocessing_keys(is_bgr, target, 'is_bgr')
```

Comment:
This test only valid when `is_bgr` etc are all dicts. Are they always dicts?

Reply:
Yes, after those lines:
https://github.com/apple/coremltools/blob/85ee6b5d1ee6c4184d4552cb4849ea53dfd842b1/coremltools/models/neural_network/builder.py#L3287

Reply:
Right...got it!!